kube_hunter/core/events/types/common.py

```diff
@@ -115,12 +120,31 @@ class NewHostEvent(Event):
     def __init__(self, host, cloud=None):
         global event_id_count
         self.host = host
-        self.cloud = cloud
+        self.cloud_type = cloud
 
         with event_id_count_lock:
             self.event_id = event_id_count
             event_id_count += 1
 
+    @cached_property
```

Comment:
Why do you use cached property?
It looks like you implement cache here

Reply:
The cached property does help here, in case get_cloud() function didn't find a cloud so event.cloud_type remains None.
if it wasn't here every time event.cloud was accessed get_cloud() will be called, which isn't what we wanted

Reply:
Your'e right but it seems that the case were cloud_type is defined never happens (please verify me).
In that case, we can drop this parameter from the constructor and then change the `cloud` property to the `get_cloud` function (with `cached_property` decorator).

Reply:
Looking at it now, i can see there are 2 places that try to set cloud_type 
they are under FromPodHostDiscovery, 2 functions called 
- azure_metadata_discovery = which returns hard-coded "Azure", we can remove that..
- traceroute_discovery = which returns pod external ip (?), we can also remove that since it's not a cloud_type..

if we do that, we can definitely do what u suggested

Reply:
We can drop the usage of `@cached_property` if we return `"NoCloud"` from `get_cloud` when cloud is not found, instead of just returning `None`.
It looks like that the output from `traceroute_discovery` should go through the `get_cloud` function.
The `azure_metadata_discovery` return value is fine, although I believe that there should be a broader logic for dealing with cloud metadata server and specifically detecting cloud type from it, but it's out of scope.